packages/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -12,17 +13,42 @@ import type { DialogProps, DialogState } from './Dialog.types';
  * @param ref - reference to root HTMLElement of Dialog
  */
 export const useDialog_unstable = (props: DialogProps, ref: React.Ref<HTMLElement>): DialogState => {
-  return {
-    // TODO add appropriate props/defaults
+  const { overlay, open = false, type = 'modal' } = props;
+  const isNonModal = type === 'non-modal';
+
+  const contentRef = React.useRef(null);
+  const { modalAttributes } = useModalAttributes({ trapFocus: !isNonModal && true });
+  const { findFirstFocusable } = useFocusFinders();
+
+  React.useEffect(() => {
+    if (open && contentRef.current) {
+      const firstFocusable = findFirstFocusable(contentRef.current);
```

Comment:
we should also allow focusing elements other than the first one. It is relatively common use case. Maybe a prop where you can pass a selector for default focusable.

Reply:
Is that a useful API ? something very similar can be done with a simple effect that would be more flexible for users because they can watch for other dependencies like waiting for loading state to end

Reply:
agree - maybe we should add a story for that then